cpp/ql/src/experimental/Security/CWE/CWE-193/InvalidPointerDeref.ql

```diff
@@ -321,7 +321,15 @@ query predicate edges(MergedPathNode node1, MergedPathNode node2) {
   or
   node1.asPathNode3().getASuccessor() = node2.asPathNode3()
   or
-  joinOn2(node1.asPathNode3(), node2.asSinkNode(), _)
+  joinOn2(node1.asPathNode3(), node2.asSinkNode(), _, _)
+}
+
+query predicate nodes(MergedPathNode n, string key, string val) {
```

Comment:
FWIW `nodes` is implicitly generated by the CLI in a `path-problem` query based on the `edges` predicate. So strictly speaking this query predicate is redundant (since we're not using it to modify the string representation of the nodes). However, I do like having the `nodes` predicate in here since it can be useful to append `concat(n.getAQlClass(), ", ")` for debugging purposes.

Reply:
Ah, learnt something new. Thanks.

Reply:
I don't blame you. I don't think this is documented anywhere 😅